src/Dmishh/Bundle/SettingsBundle/Form/Type/SettingsType.php

```diff
@@ -75,23 +76,39 @@ function ($label) use ($fieldOptions) {
     }
 
     /**
+     * SF < 2.7
      * {@inheritDoc}
+     * @deprecated since version 2.3, to be renamed in 3.0.
+     *             Use the method configureSettings instead
      */
     public function setDefaultOptions(OptionsResolverInterface $resolver)
     {
+        $this->configureOptions($resolver);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
         $resolver->setDefaults(
             array(
                 'disabled_settings' => array(),
             )
         );
 
-        $resolver->addAllowedTypes(
-            array(
-                'disabled_settings' => 'array'
-            )
-        );
+        if (method_exists($resolver, 'setDefined')) {
```

Comment:
@matthieuy thanks for the PR.
Please, add a line comment about the compatibility